File: python/src/xiaoyi/domain/token/syntax/primitive/int16/kind.py
"""Syntax primitive integer token kind: INT16.

Layer hierarchy:
- 1 syntax
- 2 primitive
- 3 int16
- 4 kind
"""

from __future__ import annotations

from enum import StrEnum


class Int16Kind(StrEnum):
    """Classification of INT16 token kinds."""

    LITERAL = "INT16_LITERAL"
    VARIABLE = "INT16_VAR"
    CAST = "INT16_CAST"

    def label(self) -> str:
        return self.value

File: rust/src/xiaoyi/domain/token/syntax/primitive/int8/kind.rs
//! Syntax primitive integer token kind: INT8.
//!
//! Layer hierarchy:
//! - 1 syntax
//! - 2 primitive
//! - 3 int8
//! - 4 kind
//!
//! Concrete variant taxonomy for INT8 syntax nodes before rendering.

#[derive(Debug, Clone, Copy)]
pub enum Int8Kind {
    Literal,
    Variable,
    Cast,
}

impl Int8Kind {
    /// Human-readable label used by parser diagnostics and codegen backends.
    pub fn label(&self) -> &'static str {
        match self {
            Int8Kind::Literal => "INT8_LITERAL",
            Int8Kind::Variable => "INT8_VAR",
            Int8Kind::Cast => "INT8_CAST",
        }
    }
}
